include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -265,6 +265,41 @@ namespace VectorTools
     }
 
 
+    template <typename VectorType>
+    typename std::enable_if<dealii::is_serial_vector<VectorType>::value ==
+                          true>::type
+            check_ghost(VectorType &vt) {}
+
+    template <typename VectorType>
+    typename std::enable_if<dealii::is_serial_vector<VectorType>::value == false
+            and std::is_base_of<BlockVectorBase<typename VectorType>,VectorType>::value == false >::type
+    check_ghost(VectorType &vt)
+    {
+        // If the vector is distributed, the following checks
+        // try to prevent problems on ghost cells
+        Assert(!vt.has_ghost_elements(),
+               ExcMessage("Vector has no write access on ghost cells"));
+
+        // Cheking for overlapping values on ghost cells:
+        // this test is not 100% accurate, but should catch most of the
+        // problems
+        const auto n_global_elements =
+          Utilities::MPI::sum(vt.local_size() + vt.n_ghost_entries(),
```

Comment:
`n_ghost_entries`  only exists for `LinearAlgebra::distributed::Vector`.

Reply:
Das ist ein großes, haariges graues Problem... just joking, thank you for pointing this out! We shall think of something...